backend/src/user/router.py

```diff
@@ -32,11 +33,17 @@ async def login_for_access_token(
     :return: JSON response containing the prices of necessities.
     :rtype: dict
     """
-    user = check_user_password_is_correct(db, form_data.username, form_data.password)
-    access_token = create_access_token(
-        data={"sub": str(user.username)}, expires_delta=timedelta(minutes=AppConfig.INTERVAL_MINUTES)
-    )
-    return {"access_token": access_token, "token_type": "bearer"}
+    try:
+        user = check_user_password_is_correct(db, form_data.username, form_data.password)
+        if not user:
+            logger.warning(f"Failed to login: Invalid credentials")
```

Comment:
The user is missing here, then you should just return a `HTTPException` and stop the function, as access token won't be successfully created with missing User object (expected Exception)
```py
 if not user:
      logger.warning(f"Failed to login: Invalid credentials")
      raise HTTPException(status_code=401, detail="User not found") # or other Exception class you defined
```